### ISSUE TITLE
quantization observers: can we relax the default epsilon value?

### ISSUE THREAD

**Issue by vkuzo** [pytorch/pytorch#90288]
### 🐛 Describe the bug

In `https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/observer.py#L208`, the epsilon value used to determine the uniform quantization scale is defined as

```
eps=torch.finfo(torch.float32).eps
```

This resolves to 1.19e-7 for float32, and the value from `finfo` is chosen so that he smallest representable number such that 1.0 + eps != 1.0.  This definition is generic might not apply well to choosing the quantization scale, I would expect the smallest uniform quantization scale value to be determined by restrictions of the underlying backend.  For example, for XNNPACK the recommended epsilon value is 2e-12.

On the pretrained weights of MobileNetV3 on ImageNet-1K from torchvision, the weights of the `features.4.block.2.fc1` and `features.4.block.2.fc2` layers are too small in magnitude for PTQ to work correctly with the current default scale value, leading to poor quantization accuracy of these layers in PTQ.

We should consider epsilon in a smarter way.  I would expect the overall impact to be low as this is an edge case, but it will be a free accuracy win for PTQ for that edge case.

### Versions

master

**Comment by Maratyszcza** [comment]
> For example, for XNNPACK the recommended epsilon value is 2e-12.

Where do you see that?

**Comment by vkuzo** [comment]
> 2e-12

https://github.com/pytorch/pytorch/pull/74396 originally added the 2e-12 epsilon for xnnpack ops in PyTorch, the PR summary has the reasoning.  Would definitely be interested in your thoughts.  

cc @digantdesai , @kimishpatel

**Comment by digantdesai** [comment]
> We should consider epsilon in a smarter way. I would expect the overall impact to be low as this is an edge case, but it will be a free accuracy win for PTQ for that edge case.

Agree. This is definitely conservative, as I said the it PR.

**Comment by Maratyszcza** [comment]
The motivation for restricting requantization scale to `>= 2**-32` is that quantized operations accumulate to 32 bits internally, and if requantization scale is smaller than that, input can't affect the output, and we'd be better of to just statically broadcast the bias tensor into the output.
